### ISSUE TITLE
Rule Change: Normalize definition identifiers in no-*-definition

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#474]
### What rule do you want to change?

no-*-definition

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```markdown
<!-- eslint markdown/no-unused-definitions: ["error", { allowDefinitions: ["MERCURY"] }] -->

[MERCURY]: https://example.com/mercury/
```

### What does the rule currently do for this code?

It reports a false positive, flagging the definition as unused, because the identifier in the allow list ("MERCURY") does not match the normalized identifier used by mdast.

### What will the rule do after it's changed?

It will correctly recognize that "MERCURY" (regardless of case or whitespace differences) matches the normalized identifier, and will not report the definition as unused.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

We can use [mdast-util-normalize-identifier](https://github.com/micromark/micromark/tree/main/packages/micromark-util-normalize-identifier), which is used by Mdast internally. This normalization issue affects no-empty-definitions, no-duplicate-definitions, and no-unused-definitions.

**Comment by nzakas** [comment]
Sounds good to me. :+1: Marking as accepted.
